tests/Doctrine/Tests/Models/Enums/FaultySwitch.php

```diff
@@ -0,0 +1,22 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Doctrine\Tests\Models\Enums;
+
+use Doctrine\ORM\Mapping\Column;
+
+class FaultySwitch
+{
+    #[Column(type: 'string')]
+    public string $value;
+
+    /**
+     * The following line is ignored on psalm and phpstan so that we can test
+     * that the mapping is throwing an exception when a non-backed enum is used.
+     *
+     * @psalm-suppress InvalidArgument
+     */
```

Comment:
Ah so this means we already have static analysis for this kind of issue. Good. I suppose it cannot hurt to also validate it at runtime.

Reply:
yes, the library is covered by phpstan and psalm
Better having a runtime check as 3rd party users might not have the static analysis in place, or the static-analysis did not run yet (like my case) :)